nixos/modules/security/acme.nix

```
@@ -111,6 +96,31 @@ let
           own server roots if needed.
         '';
       };
+
+      dnsProvider = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "route53";
+        description = "DNS Challenge provider";
+      };
+
+      credentialsFile = mkOption {
+        type = types.str;
+        description = ''
+          File containing DNS provider credentials passed as environment variables.
+          See https://go-acme.github.io/lego/dns/ for more information.
+        '';
+        example = "/var/src/secrets/example.org-route53-api-token";
+      };
+
+      dnsPropagationCheck = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Toggles LEGo DNS propagation check, which is used alongside DNS-01
```

Legos name just seems to be lowercased, see for example the documentation https://go-acme.github.io/lego/installation/.
```suggestion
          Toggles lego DNS propagation check, which is used alongside DNS-01
```